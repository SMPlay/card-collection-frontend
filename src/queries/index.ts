export { GET_CARD_COLLECTIONS } from "./get-card-collections";
export { GET_COLLECTIONS_NAME } from "./get-collections-name";
export { LOGIN } from "./login";
export { IS_AUTH } from "./is-auth";
export { GET_REFRESH_TOKEN } from './get-refresh-token';
export { REGISTRATION_QUERY } from "./registration-query";
export { CONFIRM_USER } from "./confirm-user";
