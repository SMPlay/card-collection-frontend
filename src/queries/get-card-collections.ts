import { gql } from "@apollo/client";

export const GET_CARD_COLLECTIONS = gql`
  query {
    cardCollections {
      id name imageUrl cardsCount release
    }
  }
`;
