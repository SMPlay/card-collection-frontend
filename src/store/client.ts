import { ApolloClient } from "@apollo/client";
import { createUploadLink } from "apollo-upload-client";

import { IS_AUTH } from "../queries";

import { cache } from "./cache";

const link = createUploadLink({ 
  uri: 'http://localhost:4001/graphql',
  credentials: "include"
});

export const client = new ApolloClient({
  link,
  cache,
});

client.writeQuery({
  query: IS_AUTH,
  data: {
    isAuth: false
  }
});
