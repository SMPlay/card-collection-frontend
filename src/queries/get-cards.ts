import { gql } from "@apollo/client";

export const GET_CARDS = gql`
    query cards($from: Int!, $limit: Int!, $collectionName: String!){
        cards(from: $from, limit: $limit, collectionName: $collectionName){
            id
            name
            imageUrl
            role
            rarity
            need
            have
            number

        }

    }
`;
