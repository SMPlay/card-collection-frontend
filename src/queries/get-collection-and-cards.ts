import { gql } from "@apollo/client";

export const GET_COLLECTION_AND_CARDS = gql`
    query cardCollection($id: ID!){
        cardCollection(id: $id){
            id
            name
            cardCollectionName
            imageUrl
            cardsCount
            release
            cards{
                id
                name
                imageUrl
                role
                rarity
                need
                have
                number
            }  
        }
    }
`;
