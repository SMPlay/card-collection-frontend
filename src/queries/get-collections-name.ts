import { gql } from "@apollo/client";

export const GET_COLLECTIONS_NAME = gql`
  query {
    cardCollections {
      id name
    }
  }
`;
