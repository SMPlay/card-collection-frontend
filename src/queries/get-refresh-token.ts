import { gql } from "@apollo/client";

export const GET_REFRESH_TOKEN = gql`
  mutation {
    refreshToken {
      id
    }
  }
`;
