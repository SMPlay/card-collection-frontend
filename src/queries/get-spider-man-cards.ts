import { gql } from "@apollo/client";

export const GET_SPIDER_MAN_CARDS = gql`
    query spiderManCards($from: Int!,$limit: Int,$collectionPart: Int!){
        spiderManCards(from: $from,limit: $limit, collectionPart: $collectionPart) {
            id name imageUrl type kind need have number 
        }
    }
`;
