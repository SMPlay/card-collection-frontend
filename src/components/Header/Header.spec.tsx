import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { MockedProvider } from "@apollo/client/testing";

import { Header } from "./header";
import { GET_COLLECTIONS_NAME } from "../../queries";

describe("Header component", () => {
  const mocks = [
    {
      request: {
        query: GET_COLLECTIONS_NAME,
      },
      result: {
        data: {
          cardCollections: [{ id: "test", name: "test" }],
        },
      },
    },
  ];

  it("should render", () => {
    const tree = renderer
      .create(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Router>
            <Header />
          </Router>
        </MockedProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
