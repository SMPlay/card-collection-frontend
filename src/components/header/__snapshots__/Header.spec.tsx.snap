// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component should render 1`] = `
<div
  className="makeStyles-root-1"
>
  <main
    className="makeStyles-content-2"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-5 mui-fixed MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="open drawer"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-9 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
        <h6
          className="MuiTypography-root makeStyles-title-10 MuiTypography-h6 MuiTypography-noWrap"
        >
          Главная
        </h6>
        <div
          className="MuiBox-root MuiBox-root-12"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <a
                className="makeStyles-navLink-11"
                href="/login"
                onClick={[Function]}
              >
                Войти
              </a>
            </span>
          </button>
          /
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <a
                className="makeStyles-navLink-11"
                href="/registration"
                onClick={[Function]}
              >
                Регистрация
              </a>
            </span>
          </button>
        </div>
      </div>
    </header>
    <div
      className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-13"
      style={
        Object {
          "width": 240,
        }
      }
    >
      <div
        className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-14 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
        style={
          Object {
            "visibility": "hidden",
          }
        }
      >
        <div
          className="makeStyles-drawerHeader-15"
        >
          <button
            className="MuiButtonBase-root MuiIconButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </span>
          </button>
        </div>
        <hr
          className="MuiDivider-root"
        />
        <ul
          className="MuiList-root MuiList-padding"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <div
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary makeStyles-navlink-16 MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Главная
                </span>
              </div>
            </div>
          </a>
          <a
            href="/gallery"
            onClick={[Function]}
          >
            <div
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary makeStyles-navlink-16 MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Галерея
                </span>
              </div>
            </div>
          </a>
          <a
            href="/auction"
            onClick={[Function]}
          >
            <div
              aria-disabled={false}
              className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiListItemText-primary makeStyles-navlink-16 MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Аукцион
                </span>
              </div>
            </div>
          </a>
          <div
            aria-disabled={false}
            className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <div
              className="MuiListItemText-root"
            >
              <span
                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
              >
                Коллекции
              </span>
            </div>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </ul>
        <hr
          className="MuiDivider-root"
        />
      </div>
    </div>
    <div
      className="makeStyles-drawerHeader-4"
    />
  </main>
</div>
`;
